raiden/transfer/architecture.py

```diff
@@ -164,12 +164,17 @@ class Event:
     __slots__ = ()
 
 
-class SendMessageEvent(Event):
+class EventWithLogTime(Event):
+    """ Marker for events that are given a timestamp when written to the wal. """
+
+
+class SendMessageEvent(EventWithLogTime):
     def __init__(
             self,
             recipient: Address,
             channel_identifier: ChannelID,
             message_identifier: MessageID,
+            log_time: int = None,
```

Comment:
I am looking at this and thinking ahead and i have several comments:
1. None of the places in which we instantiate events will provide this, hence the None default argument. Which makes this redundant and unnecessary (code-wise).
2. We store the `log_time` in a separate field in the database, but we also store a `log_time` field as part of the serialized events which is also redundant (data-wise).
3. Given #2, once  the deserialization happens, what you could do is "inject" the `log_time` attribute into the deserialized events. Since we don't have access modifiers in Python, it should be fine if we do this. So i would revert all the changes that you did on all events.

One more note about this which is that when we move to using `dataclasses`, the base dataclass could define the log_time field so that it's set to `None` by default and some value when deserialization takes place.

Reply:
The `to_dict` methods of the different events only add `log_time` to the dict representation if it has not been set, which should never happen. So the redundancy in the database (#2) will not occur.

I thought of 'injecting' a log_time attribute too but this approach looked safer and more readable to me. (If the API is given an event with no log-time for some reason, it should display it as such and not crash with "no such attribute"). Though another way to achieve this would be to make `log_time` a property of `Event` and read it with `getattr` internally.

Reply:
Here's how i think the implementation should have been done:
1. Introduce log_time field into the DB
2. Whenever an event is being stored, we set the timestamp field with the current time in ms.
3. When deserializing, we set the `log_time` field of those events.
4. an API call to fetch internal events will call [this function](https://github.com/raiden-network/raiden/blob/master/raiden/api/rest.py#L825) which will make sure each event returned will have the field injected.
The above means that the frontend will never receive a payload with the `log_time` field missing.

As for the redundancy in data, what i meant was the fact that we will have 2 places which may contain the same value... the `log_time` field in the DB row and the `log_time` field in the serialized event instance stored in `data` field.

Reply:
Ok. That's what I did, except that I made `log_time` a regular attribute of the `Event` subclasses. (Which would not be serialized and put in the database as `to_dict` skips it if it is None). You are right that it is unnecessary and I will remove it.